### ISSUE TITLE
Sky colormap not working properly for the pale garden biome

### ISSUE THREAD

**Issue by Emfire12345** [sp614x/optifine#7815]
Using Optifine 1.21.3_HD_U_J2_pre11

When using grid colormaps for the sky and appling a gray color to the pale garden biome (id=65) the sky isnt colored gray but instead deep turquoise but when i try to counteract the color (using a stronger color) the color is applied properly, in short, i cant get the sky to not look bad.

Color example 1:
![1](https://github.com/user-attachments/assets/9369aef7-ab90-4543-b99c-f9fb120f1c80)
In game 1:
![2024-12-01_17 21 07](https://github.com/user-attachments/assets/668dfc29-54dc-48f3-9c22-996d597e22e4)

Color example 2:
![2](https://github.com/user-attachments/assets/c06dc57c-6d83-4c6f-94ec-c350ffc96253)
In game 2:
![2024-12-01_17 29 30](https://github.com/user-attachments/assets/b1eea8f0-084e-4ad9-9b62-169927e6ca44)

**Comment by SeaOfPixels** [comment]
Can confirm in 1.21.4pre1. It's worth noting that the transition is also messed up. It fades into an incorrect orangish color when entering the biome, and then after it fully fades it switches to the green color in the above screenshots.

**Comment by Emfire12345** [comment]
> Can confirm in 1.21.4pre1. It's worth noting that the transition is also messed up. It fades into an incorrect orangish color when entering the biome, and then after it fully fades it switches to the green color in the above screenshots. Also to the OP: pale garden is ID 12, not 65.

12? Was it changed in 1.21.4? In optikai it was 0, in 1.21.3 optifine it was 65 and now it is 12? Make up your god damn mind :/

Also, i was able to make the sky almost gray by using a magenta color but yeah, transition is awfull then

**Comment by SeaOfPixels** [comment]
It's 12 because it comes after Dark Forest which is 11. Idk how it was in the experimental phase but in 1.21.4 it's 12.

**Comment by Emfire12345** [comment]
> It's 12 because it comes after Dark Forest which is 11. Idk how it was in the experimental phase but in 1.21.4 it's 12.

I would kinda prefere if they added new biome at the end of the id list... wouldnt mess up earlier version (or remap ids with an optifine feature)

**Comment by SeaOfPixels** [comment]
Yup, but at least we have pack overlays now so once you update there won't be any issues with backwards compatibility.

**Comment by Emfire12345** [comment]
> Yup, but at least we have pack overlays now so once you update there won't be any issues with backwards compatibility.

Am i missing something? We are not talking about layering packs but a single resourcepack, right?

**Comment by SeaOfPixels** [comment]
Yes, if you are unaware we can now make overlays within a single resource pack using pack.mcmeta. Here is an example of a pack.mcmeta that implements a pack overlay:
```
{
    "pack": {
        "pack_format": 46,
        "description": "§6New Default+ §3v1.79: §bMC1.20-1.21.4 §7- §9SeaOfPixels"
    },
    "overlays": {
        "entries": [
            {
                "directory": "components_1.21.4",
                "formats": {
                    "min_inclusive": 43,
                    "max_inclusive": 46
                }
            }
        ]
    }
}
```
In this example, the pack.mcmeta is specifiying that only when playing in versions with pack format 43/44/45/46, the overlay folder `components_1.21.4` should take priority over anything in `assets/minecraft`. This is desirable for making a pack backwards compatible when a breaking change occurs in a Minecraft update. The overlay folder `components_1.21.4` is put in the root folder, the same one that assets is in, just to clarify. Hope this helps, if you have further questions I recommend asking in the Optifine discord as this sort of dicussion shouldn't be happening on a github bug report.

**Comment by SeaOfPixels** [comment]
Can confirm in pre13.

**Comment by sp614x** [comment]
Can you post an example RP?

**Comment by Emfire12345** [comment]
https://www.mediafire.com/file/u22nlc0487mzd6i/Colormap_Test.zip/file

The problem is that the game shifts the skycolor for the biome, this does not only effect the pale garden but others as well, with most of them it isnt as strong however.

I found a workaround in my pack for this issue by making the color in the colormap blue, turning into grey in the game.

**Comment by sp614x** [comment]
Is this only with custom sky?
Example resource pack?

**Comment by Emfire12345** [comment]
Fog could be affected as well but in my testing it was so minor that i could have been imagining it.
So yes, only for the sky colormap.

**Comment by SeaOfPixels** [comment]
Other test resource pack: 

[NewDefault+OnlyColormapsNew.zip](https://github.com/user-attachments/files/18716653/NewDefault%2BOnlyColormapsNew.zip)

**Comment by SeaOfPixels** [comment]
Can confirm in OptiFine 1.21.4_HD_U_J3 & OptiFine 1.21.4_HD_U_J4_pre1.
